libhoney.go

```diff
@@ -471,6 +482,12 @@ func (f *fieldHolder) AddFunc(fn func() (string, interface{}, error)) error {
 	return nil
 }
 
+func (f *fieldHolder) Fields() map[string]interface{} {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	return f.data
+}
```

Comment:
Hm... leaning towards just returning a reference here, given that this method is mostly just for testing, so that you can assert on an event's contents. In that case we do not need the lock.